Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp

```diff
@@ -315,6 +508,38 @@ bool TerrainSystem::GetIsHoleFromFloats(float x, float y, Sampler sampler) const
     return !terrainExists;
 }
 
+void TerrainSystem::GetNormalsSynchronous(const AZStd::span<AZ::Vector3>& inPositions, Sampler sampler, 
+    AZStd::span<AZ::Vector3> normals, AZStd::span<bool> terrainExists) const
+{
+    AZStd::vector<AZ::Vector3> directionVectors(inPositions.size() * 4);
+    const AZ::Vector2 range(m_currentSettings.m_heightQueryResolution / 2.0f, m_currentSettings.m_heightQueryResolution / 2.0f);
+    size_t indexStepSize = 4;
+    for (size_t i = 0, iteratorIndex = 0; i < inPositions.size(); i++, iteratorIndex += indexStepSize)
+    {
+        directionVectors[iteratorIndex].Set(inPositions[i].GetX(), inPositions[i].GetY() - range.GetY(), 0.0f);
+        directionVectors[iteratorIndex + 1].Set(inPositions[i].GetX() - range.GetX(), inPositions[i].GetY(), 0.0f);
+        directionVectors[iteratorIndex + 2].Set(inPositions[i].GetX() + range.GetX(), inPositions[i].GetY(), 0.0f);
+        directionVectors[iteratorIndex + 3].Set(inPositions[i].GetX(), inPositions[i].GetY() + range.GetY(), 0.0f);
+    }
```

Comment:
Suggestion: Again consider reserve() and emplace_back to avoid constructing default Vector3s. It also simplifies the loop down to `for (auto& inPosition : inPositions)`

Reply:
Fixed